.circleci/config.yml

```diff
@@ -72,6 +72,38 @@ commands:
           key: npm-cache
           paths:
             - /root/.npm
+  setup_npm_7:
+    description: 'Setup Machine with npm 7'
+    steps:
+      - restore_cache:
+          keys:
+            - npm-cache
+      - run:
+          name: 'Install Dependencies'
+          command: |
+
+            sudo npm install -g npm@7
+            npm ci
+      - save_cache:
+          key: npm-cache
+          paths:
+            - /root/.npm
```

Comment:
I'm no CircleCI experts, but I wonder does the different npm jobs defeat the purpose of the cache...

In this circleci config, I see a few `restore_cache` using the same key `npm-cache`...

My naive understanding is, we install dependencies in the npm6 job, then we save the dependencies in the cache under the key `npm-cache`, and then we have 2 more jobs (npm6 and npm7), and would they invalidate the cache?

Reply:
one way we can find out is to run the circle ci tests a few times in a row, and we compare the time it takes for CircleCi to install the dependencies.

Reply:
@kevinxh I ran the tests a few times and the time it takes to install dependencies is consistently around 2 min 50 sec on the main node 14 job while the node 14 npm 7, npm 8, and windows builds all fluctuate between 1:30 to ~1:50. It looks like the cache isn't being invalidated and that it is shaving off about a minute in the latter 3 jobs.